src/DssSpell.t.base.sol

```diff
@@ -1839,48 +1814,175 @@ contract DssSpellTestBase is Config, DssTest {
+        string[] memory dkeys = new string[](3);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[4] memory keys) internal {
+        string[] memory dkeys = new string[](4);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[5] memory keys) internal {
+        string[] memory dkeys = new string[](5);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        dkeys[4] = keys[4];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[6] memory keys) internal {
+        string[] memory dkeys = new string[](6);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        dkeys[4] = keys[4];
+        dkeys[5] = keys[5];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[7] memory keys) internal {
+        string[] memory dkeys = new string[](7);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        dkeys[4] = keys[4];
+        dkeys[5] = keys[5];
+        dkeys[6] = keys[6];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[8] memory keys) internal {
+        string[] memory dkeys = new string[](8);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        dkeys[4] = keys[4];
+        dkeys[5] = keys[5];
+        dkeys[6] = keys[6];
+        dkeys[7] = keys[7];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[9] memory keys) internal {
+        string[] memory dkeys = new string[](9);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        dkeys[4] = keys[4];
+        dkeys[5] = keys[5];
+        dkeys[6] = keys[6];
+        dkeys[7] = keys[7];
+        dkeys[8] = keys[8];
+        _testNewOrUpdatedChainlogValues(dkeys);
+    }
+
+    function _testNewOrUpdatedChainlogValues(string[10] memory keys) internal {
+        string[] memory dkeys = new string[](10);
+        dkeys[0] = keys[0];
+        dkeys[1] = keys[1];
+        dkeys[2] = keys[2];
+        dkeys[3] = keys[3];
+        dkeys[4] = keys[4];
+        dkeys[5] = keys[5];
+        dkeys[6] = keys[6];
+        dkeys[7] = keys[7];
+        dkeys[8] = keys[8];
+        dkeys[9] = keys[9];
+        _testNewOrUpdatedChainlogValues(dkeys);
```

Comment:
this looks slightly inelegant imho :/

Reply:
I removed the overloads.

Unfortunately, Solidity is quite limited here, so there's no good syntactic sugar to make the call cleaner. This was an attempt to make frequent changes simpler.